### PR TITLE
switch GCR -> Artifact-Registry

### PR DESCRIPTION
**Pull request by ccwienk**
GCR has been deprecated [0] in favour of Artifact-Registry.

Thus, change push-targets for OCI-Images:

- europe-docker.pkg.dev/gardener-project/snapshots for snapshots
- europe-docker.pkg.dev/gardener-project/releases for releases
- europe-docker.pkg.dev/gardener-project/public for combined view of snapshots + releases

[0]
https://cloud.google.com/artifact-registry/docs/transition/transition-from-gcr

**Release note**:

```breaking operator
Change OCI Image Registry from GCR (`eu.gcr.io/gardener-project`) to Artifact-Registry (`europe-docker.pkg.dev/gardener-project/releases`). Users should update their references.

```

### CODE REVIEW

**Review by vpnachev — COMMENTED** [review]
For most of the changes I cannot provide meaningful feedback because the pipelines are quite outside of my scope. As there are similar changes, I would prefer to put on hold this change for some time and wait the other components to go through each pipeline option (including the release) and merge this change just after some experience is gained.

/hold

**Comment by ccwienk** [comment]
@vpnachev : by now, almost all similar pullrequests have been merged, and there were already a couple of releases to Artifact-Registry instead of GCR (yesterday, the largest change to [gardener/gardener](https://github.com/gardener/gardener/pull/8968) was also accepted. I think it might now be safe to proceed (note: after the next gardener-release, revendoring of Gardener will likely make _this_ pull-request mandatory due to changes in `hack/.ci/component_descriptor`).

**Comment by vpnachev** [comment]
@ccwienk, in the light of https://github.com/gardener/gardener/pull/9008, does this PR require adaptation similar to https://github.com/gardener/gardener-extension-provider-azure/pull/770?

**Comment by ccwienk** [comment]
@vpnachev : this change should not be necessary, considering this repository does not seem to make use of the special-handling of `images.yaml`

**Review by vpnachev — APPROVED** [review]
/lgtm

**Pull request merged**
